omegaconf/dictconfig.py

```diff
@@ -662,6 +672,9 @@ def _dict_conf_eq(d1: "DictConfig", d2: "DictConfig") -> bool:
         assert isinstance(d2, DictConfig)
         if len(d1) != len(d2):
             return False
+        if d1._is_missing() or d2._is_missing():
+            return d1._is_missing() == d2._is_missing()
```

Comment:
Very minor optimization (I doubt we care, but since we've been hunting down bottlenecks...)
```suggestion
            return d1._is_missing() and d2._is_missing()
```